internal/controllers/users.go

```diff
@@ -0,0 +1,171 @@
+	}
+	return ctx.JSON(http.StatusOK, model.SuccessResponse(plandata, http.StatusOK))
+}
+
+// swagger:route GET /users/{UserName}/quotas users listAllUserQuotaByID
+// Returns a Lists the User Quota Details.
+// responses:
+//   200: UserResponse
+//   404: RootResponse
+//
+func (s Server) GetUserAllQuotas(ctx echo.Context) error {
+	resource := ctx.QueryParam("resource")
+	resourceFilter := ""
+	if resource != "" {
+		resourceFilter = `and resource_types.name=?`
+	}
+	username := ctx.Param("username")
+	if username == "" {
+		return ctx.JSON(http.StatusBadRequest, model.ErrorResponse("Invalid Usernaem", http.StatusBadRequest))
+	}
+
+	plandata := []QuotaDetails{}
+	err := s.GORMDB.Raw(`select plans.name as plan_name,quotas.quota,resource_types.name as resource_nam,resource_types.unit as resource_unit from
+	user_plans
+	join plans on plans.id=user_plans.plan_id
+	join quotas on user_plans.id=quotas.user_plan_id
+	join resource_types on resource_types.id=quotas.resource_type_id
+	join users on users.id=user_plans.user_id
+	where
+	users.username=?`+resourceFilter, username).Scan(&plandata).Error
+	if err != nil {
+		return ctx.JSON(http.StatusBadRequest, map[string]interface{}{"error": err.Error()})
+	}
+	return ctx.JSON(http.StatusOK, model.SuccessResponse(plandata, http.StatusOK))
+}
+
+// swagger:route GET /users/{UserName}/quotas/{quotaid} users listUserQuotaByID
+// Returns a Lists the User Quota Details.
+// responses:
+//   200: UserResponse
+//   404: RootResponse
+//
+func (s Server) GetUserQuotaDetails(ctx echo.Context) error {
+	quotaid := ctx.Param("quotaid")
+	if quotaid == "" {
+		return ctx.JSON(http.StatusBadRequest, model.ErrorResponse("Invalid quota Id", http.StatusBadRequest))
+	}
+
+	username := ctx.Param("username")
+	if username == "" {
+		return ctx.JSON(http.StatusBadRequest, model.ErrorResponse("Invalid Usrname", http.StatusBadRequest))
+	}
+
+	plandata := []PlanDetails{}
+	err := s.GORMDB.Raw(`select plans.name as plan_name,quotas.quota,resource_types.name as resource_nam,resource_types.unit as resource_unit from
+	user_plans
+	join plans on plans.id=user_plans.plan_id
+	join quotas on user_plans.id=quotas.user_plan_id
+	join resource_types on resource_types.id=quotas.resource_type_id
+	join users on users.id=user_plans.user_id
+	where
+	users.username=? and quotas.id=?`, username, quotaid).Scan(&plandata).Error
+	if err != nil {
+		return ctx.JSON(http.StatusBadRequest, map[string]interface{}{"error": err.Error()})
+	}
+	return ctx.JSON(http.StatusOK, model.SuccessResponse(plandata, http.StatusOK))
+}
+
+// swagger:route GET /users/{UserName}/usages users listUserUsageDetailsByID
+// Returns a Lists the User Quota Details.
+// responses:
+//   200: UserResponse
+//   404: RootResponse
+//
+func (s Server) GetUserUsageDetails(ctx echo.Context) error {
+	resource := ctx.QueryParam("resource")
+	resourceFilter := ""
+	if resource != "" {
+		resourceFilter = `and resource_types.name=?`
+	}
```

Comment:
Something like this is where things like [Masterminds/squirrel](https://github.com/Masterminds/squirrel) can really come in handy. I haven't used GORM, much, but I wonder if GORM has a nice way to conditionally add filters like this.